src/RoadRegistry.BackOffice/Core/ModifyRoadSegment.cs

```diff
@@ -346,16 +346,15 @@ public Problems VerifyAfter(AfterVerificationContext context)
                 }
             }
 
-            // TODO: WR-290 re-enable
-            // var intersectingSegments = context.AfterView.View.CreateScopedView(Geometry.EnvelopeInternal)
-            //     .Segments.Where(pair => pair.Key != Id && pair.Value.Geometry.Intersects(Geometry));
-            // var intersectingSegmentsWithoutJunction = intersectingSegments.Where(intersectingSegment =>
-            //     !context.AfterView.GradeSeparatedJunctions.Any(junction =>
-            //         (junction.Value.LowerSegment == Id && junction.Value.UpperSegment == intersectingSegment.Key) ||
-            //         (junction.Value.LowerSegment == intersectingSegment.Key && junction.Value.UpperSegment == Id)));
-            //
-            // problems = problems.AddRange(intersectingSegmentsWithoutJunction.Select(i =>
-            //     new IntersectingRoadSegmentsDoNotHaveGradeSeparatedJunction(Id, i.Key)));
+            var intersectingSegments = context.AfterView.View.CreateScopedView(Geometry.EnvelopeInternal)
+                .Segments.Where(pair => pair.Key != Id && pair.Value.Geometry.Intersects(Geometry) && !pair.Value.Nodes.Any(node => new [] {StartNodeId, EndNodeId}.Contains(node)));
```

Comment:
is dit niet dezelfde logica als in AddRoadSegment.cs? indien ja dan zie ik dit heel graag in 1 en dezelfde method ergens

Reply:
Goed opgemerkt, ik heb het samengestoken in [ScopedRoadNetworkViewExtensions](https://github.com/Informatievlaanderen/road-registry/pull/392/files#diff-d02d36f2e69ba5a4f4d18464b36171e2648dc50ea31d43374494bb16cd60853a)